src/text/non_blocking/raw_text_reader.rs

```diff
@@ -37,8 +46,13 @@ pub struct RawTextReader<A: AsRef<[u8]>> {
     // Otherwise, it is None.
     current_ivm: Option<(u8, u8)>,
     bytes_read: usize,
+    // True if the current text buffer is exhausted.
     is_eof: bool,
+    // True if the caller has indicated that all data has been read for the 'stream'.
+    is_eos: bool,
     parents: Vec<ParentContainer>,
+    // Optional state from previous read attempts.
+    carried_state: Option<ReaderState>,
```

Comment:
I think we can merge a couple of these member fields; rather than holding a `carried_state: Option<ReaderState>`, I think it can have a `state: ReaderState` with a new variant that represents "ready to read the next value".

See the non-blocking raw binary reader's [`ReaderState` type](https://github.dev/amzn/ion-rust/blob/b01a603216edfe83d51aa57647ea912665f5334b/src/binary/non_blocking/raw_binary_reader.rs#L233-L261) for comparison.